### PR TITLE
Fe auto redirect

### PR DESCRIPTION
**Pull request by ruslan-akhm**
- When unauthenticated user tries to access protected route they are redirected to login page with their previous locatioin saved
- Once logged in user brought to previous location they tried to access
- 404 page was added as well

### CODE REVIEW

**Review by dproc96 — CHANGES_REQUESTED** [review]
This is really great! I love the 404 page and how simple your solution for redirecting is!

One change to make, can you move your inline styles out into the makeStyles function and use them as classes? It will just make the code more consistent and readable

**Comment by ruslan-akhm** [comment]
> This is really great! I love the 404 page and how simple your solution for redirecting is!
> 
> One change to make, can you move your inline styles out into the makeStyles function and use them as classes? It will just make the code more consistent and readable

Just did

File: client/src/pages/PageNotFound.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  const classes = useStyles();
+  return (
+    <Grid
+      container
+      className={classes.parentBox}
+      justify="center"
+      alignItems="center"
+      direction="column"
+    >
+      <Grid item style={{ marginTop: "auto" }}>
```
Comment (dproc96) [inline client/src/pages/PageNotFound.js]: Can you move all of these style out to the makeStyles?

Comment (ruslan-akhm) [reply client/src/pages/PageNotFound.js]: Ok. Everything that could be moved into classes - was moved into classes now

File: client/src/pages/PageNotFound.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+          <Link to="/dashboard/profile" className={classes.profileLink}>
+            Back to profile
+          </Link>
+        </Typography>
+      </Grid>
+      <Grid item>
+        <Paper
+          square
+          elevation={0}
+          style={{}}
```
Comment (dproc96) [inline client/src/pages/PageNotFound.js]:
```suggestion
```
This can be removed now

**Review by dproc96 — APPROVED** [review]
Spotted one empty style property, you can remove that and then you're good to merge!

File: client/src/pages/PageNotFound.js (hunk 2 above)
Comment (ruslan-akhm) [reply client/src/pages/PageNotFound.js]: Oh...somehow missed it. Fixed now

**Pull request merged**